plugin/evm/vm.go

```diff
@@ -379,9 +379,15 @@ func (vm *VM) Initialize(
 		vm.chainConfig.UpgradeConfig = upgradeConfig
 	}
 
-	// create genesisHash after applying upgradeBytes in case
+	// create genesisBlock after applying upgradeBytes in case
 	// upgradeBytes modifies genesis.
-	vm.genesisHash = vm.ethConfig.Genesis.ToBlock(nil).Hash()
+	genesisBlock := vm.ethConfig.Genesis.ToBlock(nil)
+	vm.genesisHash = genesisBlock.Hash()
+
+	// check that subnetEVM upgrade is enabled from genesis
+	if !vm.chainConfig.IsSubnetEVM(genesisBlock.Number()) {
```

Comment:
this should be timestamp or hardcoded to 0

Reply:
thanks